File: 001/erlang/one.erl
-module(one).
-export([main/0, sum/1, sum/2, mod/2]).

mod(X,Y)->(X rem Y + Y) rem Y.
sum(L) -> sum(L,0).
sum([], Sum) -> Sum;
sum([H|T], Sum) -> sum(T, H+Sum).

main() ->
    io:format("~p\n", [sum(
        lists:filter(
            fun (X) -> (mod(X, 3) == 0) or (mod(X, 5) == 0) end,
            lists:seq(1,999)
        )
    )]).

File: 001/fsharp/one.fs
let divisibleBy3or5 seq =
   let isDivisibleBy3or5 x = x%3 = 0 || x%5 = 0
   Seq.sum(Seq.filter isDivisibleBy3or5 seq)

printfn "%d" (divisibleBy3or5(seq { 0 .. 999 }))
